### ISSUE TITLE
[bug] use of replace_requires to update version for the same recipes break in 2.10

### ISSUE THREAD

**Issue by hwwan80** [conan-io/conan#17407]
### Describe the bug

conan version: 2.10

conan profile:
Host profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0

Build profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0

### How to reproduce it

```
conan install --requires=graphene/1.10.8  --build=missing

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0


======== Computing dependency graph ========
Graph root
    cli
Requirements
    bzip2/1.0.8#d00dac990f08d991998d624be81a9526 - Cache
    glib/2.81.0#70a4237610c8f5e426882afab6d8fd92 - Cache
    graphene/1.10.8#8f0e385bc9e23eba47c2d1697c3f512a - Cache
    libelf/0.8.13#4f70a3555809ae50cc8add44f0f84288 - Cache
    libffi/3.4.4#72cccac6139293b6e97ec9cd676547ff - Cache
    libmount/2.39.2#64850679fa9d5ead878e5076b0cab2a8 - Cache
    libselinux/3.6#5a78ff6ae5034eeaac8da723361a8ce4 - Cache
    pcre2/10.42#ac2227071f79a062167cf2fee7189d66 - Cache
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache
Build requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809 - Cache
    flex/2.6.4#e35bc44b3fcbcd661e0af0dc5b5b1ad4 - Cache
    gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
    meson/1.2.2#21b73818ba96d9eea465b310b5bbc993 - Cache
    meson/1.3.1#a17179442f5a99378b58ef0a0f43babc - Cache
    meson/1.3.2#26ce8a76a36cc275cdfee1d757bc6561 - Cache
    ninja/1.12.1#fd583651bf0c6a901943495d49878803 - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
    pkgconf/2.2.0#6462942a22803086372db44689ba825f - Cache
Replaced requires
    glib/2.78.3: glib/2.81.0
Resolved version ranges
    meson/[>=1.2.3 <2]: meson/1.3.2
    ninja/[>=1.10.2 <2]: ninja/1.12.1
    pkgconf/[>=2.2 <3]: pkgconf/2.2.0
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
ERROR: graphene/1.10.8: Error in validate() method, line 62
	glib_is_shared = self.dependencies["glib"].options.shared
	KeyError: "'glib' not found in the dependency set"

```

It worked in conan 2.9.3

**Comment by cynix** [comment]
I’m also seeing this, as well as other problems on 2.10. Even if `conan install` finishes without error, a lot of the expected .cmake files are missing from the generator output.

**Comment by AbrilRBS** [comment]
Hi @hwwan80 @cynix thanks a lot for your report. This indeed seems like a bug in the newly released 2.10 version seemingly introduced in https://github.com/conan-io/conan/pull/17326.

I'm proposing https://github.com/conan-io/conan/pull/17409 to fix this, which will probably be released as 2.10.1 soon, thanks a lot for taking the time to report this issue :)

**Comment by memsharded** [comment]
This has been closed by https://github.com/conan-io/conan/pull/17409, it will be released in Conan 2.10.1 asap, please update when it is out and let us know. Many thanks again for reporting!
